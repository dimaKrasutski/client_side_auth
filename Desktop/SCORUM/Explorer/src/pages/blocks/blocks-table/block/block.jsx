import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { wrapRoute2Locale } from 'theme/helpers/localization';
import { toStandardFormat } from '../../../../helpers/date';
import './block.scss';

export class BlockUI extends Component {
  static propTypes = {
    history: PropTypes.shape().isRequired,
    locale: PropTypes.string.isRequired,
    block: PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.shape({
          timestamp: PropTypes.string.isRequired,
          block_id: PropTypes.string.isRequired,
        }),
      ])
    ).isRequired,
  };

  openBlockDetailsPage = () => {
    const { locale, history, block } = this.props;
    const [height] = block;
    history.push(wrapRoute2Locale(`/block-info/${height}`, locale));
  };

  render() {
    const { locale, block } = this.props;

    // TODO: rewrite
    const [height, blockData] = block;

    const {
      timestamp,
      block_id: blockId,
      transactions,
    } = blockData;

    return (
      <tr className="block" onClick={this.openBlockDetailsPage}>
        <td>
          {height}
        </td>
        <td>
          <span>{toStandardFormat(timestamp, locale)}</span>
        </td>
        <td>
          {blockId}
        </td>
        <td className="text-right">
          {transactions.length}
        </td>
      </tr>
    );
  }
}

export const Block = withRouter(BlockUI);
