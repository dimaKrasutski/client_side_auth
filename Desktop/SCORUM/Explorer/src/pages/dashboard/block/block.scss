.dashboard-block-line {
  cursor: pointer;
}
