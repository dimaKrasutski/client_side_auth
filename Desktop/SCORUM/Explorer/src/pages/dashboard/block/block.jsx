import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { wrapRoute2Locale } from '@scorum/theme-bootstrap/src/helpers/localization';
import { toStandardFormat } from '../../../helpers/date';
import './block.scss';

export class BlockUI extends Component {
  static propTypes = {
    history: PropTypes.shape().isRequired,
    locale: PropTypes.string.isRequired,
    block: PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.shape({
          timestamp: PropTypes.string.isRequired,
          block_id: PropTypes.string.isRequired,
        }),
      ])
    ).isRequired,
  };

  openBlockDetailsPage = () => {
    const { locale, history, block } = this.props;
    const [height] = block;
    history.push(wrapRoute2Locale(`/block-info/${height}`, locale));
  };

  render() {
    const { locale, block } = this.props;

    // TODO: rewrite
    const [height, blockData] = block;

    const {
      timestamp,
      block_id: blockId,
    } = blockData;

    return (
      <tr className="dashboard-block-line" onClick={this.openBlockDetailsPage}>
        <td>
          {height}
        </td>
        <td>
          <span>{toStandardFormat(timestamp, locale)}</span>
        </td>
        <td>
          {blockId}
        </td>
      </tr>
    );
  }
}

export const Block = withRouter(BlockUI);
