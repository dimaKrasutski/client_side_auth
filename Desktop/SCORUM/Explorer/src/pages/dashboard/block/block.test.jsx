import React from 'react';
import { BlockUI } from './block';

describe('<BlockUI />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<BlockUI
      block={[100, { timestamp: '2018-04-06T05:55:51', block_id: 123 }]}
      locale="en-us"
      history={{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
