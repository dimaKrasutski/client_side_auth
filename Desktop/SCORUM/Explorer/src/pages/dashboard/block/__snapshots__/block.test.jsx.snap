// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockUI /> should match snapshot 1`] = `
<tr
  className="dashboard-block-line"
  onClick={[Function]}
>
  <td>
    100
  </td>
  <td>
    <span>
      06/04/18 5:55 AM
    </span>
  </td>
  <td>
    123
  </td>
</tr>
`;
